src/plugins/data/public/query/query_string/dataset_service/lib/index_type.ts

```diff
@@ -120,11 +120,15 @@ const fetchDataSources = async (client: SavedObjectsClientContract) => {
     perPage: 10000,
   });
   const dataSources: DataStructure[] = [DEFAULT_DATA.STRUCTURES.LOCAL_DATASOURCE].concat(
-    response.savedObjects.map((savedObject) => ({
-      id: savedObject.id,
-      title: savedObject.attributes.title,
-      type: 'DATA_SOURCE',
-    }))
+    response.savedObjects
```

Comment:
i think this file has tests as well we can add which should be copy pasta from the other one

Reply:
but it can be another PR